lib/ancestry/instance_methods.rb

```diff
@@ -288,16 +284,21 @@ def ancestry_callbacks_disabled?
 
   private
 
-    def cast_primary_key(key)
-      if [:string, :uuid, :text].include? primary_key_type
-        key
-      else
-        key.to_i
-      end
+    def parse_ancestry_column obj
+      cast_primary_keys(obj.to_s.split('/'))
+    end
+
+    def cast_primary_keys(keys)
+      skip_cast? ? keys : keys.map(&:to_i)
```

Comment:
Can this be a `.map!` to avoid allocating a new array?

Reply:
If you decide to do the above, maybe change the name of the method to `cast_primary_keys!` to denote that it mutates that array.